workspaces/adr/plugins/adr-backend/src/service/router.ts

```diff
@@ -133,5 +147,75 @@ export async function createRouter(
     }
   });
 
+  router.get('/image', async (req, res) => {
+    const { token } = await auth.getPluginRequestToken({
+      onBehalfOf: await auth.getOwnServiceCredentials(),
+      targetPluginId: 'adr',
+    });
+
+    const urlToProcess = req.query.url as string;
+    if (!urlToProcess) {
+      res.statusCode = 400;
+      res.json({ message: 'No URL provided' });
+      return;
+    }
+
+    const imageTypeMap: Record<string, string> = {
+      png: 'image/png',
+      jpg: 'image/jpeg',
+      jpeg: 'image/jpeg',
+      gif: 'image/gif',
+      svg: 'image/svg+xml',
+      webp: 'image/webp',
+    };
+
+    const imageType = urlToProcess.match(/\.([a-z0-9]+)(\?.*)?$/i);
+    if (!(imageType && imageType[1])) {
+      res.statusCode = 400;
+      res.json({ message: 'No URL to image' });
+      return;
+    }
+    if (!imageTypeMap[imageType[1].toLowerCase()]) {
+      res.statusCode = 400;
+      res.json({ message: `Image type ${imageType[1]} is not supported` });
+      return;
+    }
+    const contentType = imageTypeMap[imageType[1].toLowerCase()];
+
+    const cachedFileContent = (await cacheClient.get(urlToProcess)) as {
+      data: string;
+      etag: string;
+    };
+
+    try {
+      const fileGetResponse = await reader.readUrl(urlToProcess, {
+        etag: cachedFileContent?.etag,
+        token: token,
```

Comment:
I don't believe you need to pass in the token here - this a internal Backstage user token which wouldn't be applicable for reading from external locations. UrlReader should be able to auth appropriately based on the currently configured integrations.

Reply:
You are right, I don't know why but when I was testing my changes on my local instance it was failing without it on Monday. And after your comment I removed it and tested it again and it worked. :thinking: 
Probably I had some messed up local env or something.

Thank you for pointing that out. I removed it and it works :)